src/heap.rs

```diff
@@ -186,20 +273,245 @@ impl PartialEq for Heaps {
     }
 }
 
+#[derive(Default, Clone, Debug)]
```

Comment:
It's technically sound, but I don't like that `Clone` is required here; AFAIU, it's only used for testing purposes when cloning VM `State` to check whether it's correctly rolled back later. Semantically, it makes little sense to clone the page pool. Could be dealt with later.

Reply:
This is correct. I don't like it either